server/tracedb/client/client.go

```diff
@@ -0,0 +1,99 @@
+package client
+
+import (
+	"context"
+
+	"github.com/kubeshop/tracetest/server/config"
+	"github.com/kubeshop/tracetest/server/model"
+	"github.com/kubeshop/tracetest/server/tracedb/connection"
+	"google.golang.org/grpc"
+)
+
+type SupportedClients string
+
+var (
+	GRPC SupportedClients = "grpc"
+	HTTP SupportedClients = "http"
+)
+
+type HttpCallback func(ctx context.Context, traceID string, client HttpClient) (model.Trace, error)
+type GrpcCallback func(ctx context.Context, traceID string, connection *grpc.ClientConn) (model.Trace, error)
+
+type Callbacks struct {
+	GRPC GrpcCallback
+	HTTP HttpCallback
+}
+
+type Client struct {
```

Comment:
This naming is difficult to understand because we are referencing as `client.Client`, which feels like "this is a client for what"? The idea behind it is here: https://github.com/golang/go/wiki/CodeReviewComments#package-names

At first glance, it seems to me something like `TraceDataSource`. Invoking other golang devs here: @schoren @mathnogueira , it makes sense folks?

Reply:
I like `TraceDataSource` or just `DataSource` I'll give it a go, thanks for pointing me to that documentation file ❤️